### PR TITLE
Correctly handle what are now 404 Not Found in Jörmungandr client

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#1049 

# Overview

- [x] I have added and reworked Jormungandr client "negative tests" to correctly capture:
   - GET endpoint fetching an unknown resource (expected error -> error at the value level)
   - GET endpoint fetching resourced using an invalid id (unexpected -> throw)

- [x] This for `getAccountState`, `getBlock` and `getDescendantIds`

- [x] I have unified `mkAccountId` and `unsafePoolId` functions in one `unsafeFromText` function.

- [x] Reworked `FromText` and `ToText` instances for `Hash purpose` for more consistency and correctness across the codebase.

# Comments

### CODE REVIEW

File: lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs [hunk 1]
```diff
@@ -258,35 +260,41 @@ spec = do
                 shouldThrow io $ \(ErrUnexpectedNetworkFailure link _) ->
                     show link == show (safeLink api (Proxy @GetTipId))
 
-    describe "White-box error path tests" $
+    describe "PATATE White-box error path tests" $
```
Comment (piotr-iohk) [inline lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs]: :potato:

Comment (KtorZ) [reply lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs]: Whoops.

Comment (KtorZ) [reply lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs]: Fixed :no_mouth:

**Review by piotr-iohk — APPROVED** [review]

**Comment by KtorZ** [comment]
Wow. I started to pull a thread that ended up being longer than expected. 
At least, we now handle parsing of hashes more consistently o/ ....

**Review by rvl — APPROVED** [review]
OK

**Pull request merged**